### PR TITLE
fix(git-hooks): make hooks robust to pull/push failures (follow-up to #53)

### PR DESCRIPTION
**Pull request by charliie-dev**
Follow-up to #53. Makes the git hooks (and the Android push script) robust to the failure modes that previously failed **silently**, without changing the core "commit even if the sync fails so your edits are never lost" behaviour.

## Background

In Logseq's auto-commit flow, `pre-commit` always runs while the working tree has uncommitted edits (that edit is *why* a commit is happening). So if the file you just edited also changed on the remote, `git pull` inside `pre-commit` refuses to merge ("Your local changes would be overwritten by merge — Aborting"). The old hook ignored that, committed anyway, diverged from the remote, and then `post-commit`'s push was rejected — with no explanation to the user. Reproduced this end-to-end before/after.

## Changes (4 commits)

**`fix(git-hooks): push current branch instead of hardcoded main`** (`post-commit`)
- `git push origin main` → `git push origin HEAD`. Repos whose default branch is `master` were silently failing to push on every commit.
- On push failure, print an actionable "run `git pull` to reconcile" hint.

**`fix(git-hooks): surface git pull failures and avoid hangs`** (`pre-commit`)
- Capture `git pull` stdout+stderr **and exit code**; on failure print a clear, actionable message. Still commits (edits stay safe locally) — just no longer silent.
- Drop the locale-fragile `"Already up to date."` English-only string match; stay quiet on success instead (no per-commit notification spam).
- `GIT_TERMINAL_PROMPT=0` so an HTTPS credential prompt **fails fast instead of hanging** Logseq's non-interactive subprocess.
- Existing non-worktree and merge/rebase-in-progress guards are unchanged.

**`fix(git-hooks): drop BatchMode GIT_SSH_COMMAND override`** (`pre-commit`, `post-commit`)
- An earlier revision also set a `BatchMode` `GIT_SSH_COMMAND` default. Dropped it after review: `GIT_TERMINAL_PROMPT=0` already covers the hang case, an SSH passphrase can't be entered in a TTY-less subprocess regardless, and the override risked silently clobbering a user's own `GIT_SSH_COMMAND` / `core.sshCommand`.

**`fix(android): keep push-graph pushing when nothing to commit`** (`Android-scripts/push-graph.sh`)
- A no-op `git commit` exits non-zero and aborted the script before `git push`, stranding already-committed work. Only commit when something is staged.
- Guard `cd` (`|| exit 1`) so it never runs git in the wrong directory.

## Verification
- `shellcheck -s sh` clean on both hooks; `push-graph.sh` SC2164 fixed (SC1091 is just the external `source`).
- **Quiet success path:** remote changed a different file → `pre-commit` pulls & merges cleanly, no output, commit + push succeed.
- **Loud failure path:** remote + local edit the same file → `pre-commit` pull aborts → prints the actionable warning, still commits locally (verified: **no conflict markers committed**), `post-commit` push rejected → prints the reconcile hint.
- **Branch fix:** confirmed `post-commit` pushes correctly to a repo whose default branch is `master`.

No behaviour change on the happy path; failures are now visible and actionable instead of silent.

### CODE REVIEW

File: Android-scripts/push-graph.sh [hunk 1]
```diff
@@ -1,6 +1,11 @@
 #!/usr/bin/bash
 source bin/source-ssh-agent
-cd storage/documents/Diario
+# Abort if the graph folder isn't where we expect, instead of running git in
+# the wrong directory.
+cd storage/documents/Diario || exit 1
```
[the remark this answers is not in the document]
Comment (charliie-dev) [reply Android-scripts/push-graph.sh]: Good call — addressed in `c786676`. `cd` now prints an actionable reason to stderr before exiting, which shows up in the Termux:Widget output:

```sh
cd storage/documents/Diario || { echo "push-graph: cannot cd to storage/documents/Diario, aborting" >&2; exit 1; }
```

File: git-hooks/pre-commit [hunk 2]
```diff
@@ -29,15 +34,28 @@ if [ -f "${GIT_DIR}/MERGE_HEAD" ] || [ -d "${GIT_DIR}/rebase-merge" ] || [ -d "$
...
+    # The most common failure in the Logseq flow: the file you just edited
+    # also changed on the remote, so git refuses to merge over your
+    # uncommitted changes ("Your local changes would be overwritten by
+    # merge"). We still let this commit through so your edits are never lost,
+    # but the subsequent push will be rejected until you reconcile. Make that
+    # loud and actionable rather than silent.
+    echo "pre-commit: 'git pull' failed (exit ${rc}). Committing LOCALLY only."
+    echo "pre-commit: your edits are safe, but the push will be rejected until you reconcile."
+    echo "pre-commit: open a terminal in your graph folder, run 'git pull', resolve any"
+    echo "pre-commit: conflicts, then let Logseq commit again (or push from the terminal)."
```
[the remark this answers is not in the document]
Comment (charliie-dev) [reply git-hooks/pre-commit:57]: Agreed — addressed in `7304ddf`. Softened both the message and the nearby comment: a failed pull doesn't always mean a *rejected* push (auth/network failures fail differently), so it now reads "your edits are safe, but the next push will likely fail until you reconcile."

File: git-hooks/pre-commit [hunk 3]
```diff
@@ -29,15 +34,28 @@ if [ -f "${GIT_DIR}/MERGE_HEAD" ] || [ -d "${GIT_DIR}/rebase-merge" ] || [ -d "$
     exit 0
 fi
 
-output=$(git pull --no-rebase)
+# Pull remote changes before committing. Capture stdout+stderr together with
+# the exit code so we can react to a failed pull instead of silently
+# committing on top of a broken sync.
+output=$(git pull --no-rebase 2>&1)
+rc=$?
```
[the remark this answers is not in the document]
Comment (charliie-dev) [reply git-hooks/pre-commit:44]: Added `--no-edit` in `7304ddf` as defensive hardening. One clarification with evidence, though: in Logseq's actual no-TTY subprocess, `git pull` does **not** open an editor or hang — when stdin isn't a terminal, git's merge auto-edit is skipped and the merge completes with the default message. I verified this with a watchdog test (merge-commit pull, `TERM=dumb`, stdin `</dev/null`) — it exited 0 in well under a second.

So `--no-edit` isn't fixing a hang in the Logseq path; its real value is the edge case where the hook is run **with** a terminal attached (e.g. a manual `git commit` while the branch has diverged), where it prevents an unexpected editor pop-up. Worth keeping for that reason.
